### ISSUE TITLE
Report Syntax Errors in JSDocs Type Annotations

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#13371]
From https://github.com/Microsoft/vscode/issues/18229

**TypeScript Version:**  2.2.0-dev.20170109

**Code**
In a JS file:

```ts
/**
 * @param {(x)=>void} x 
 **/
function foo(x) { }
```

**Expected behavior:**
Reports a syntax error when trying to parse the JSDoc type annotation 

**Actual behavior:**
No error reported. `x` has `any` type

**Comment by mhegazy** [comment]
Also related to https://github.com/Microsoft/TypeScript/issues/6802

**Comment by andy-ms** [comment]
This is more important now that we have `@ts-check`. Although turning on `--noImplicitAny` will tell you when you got a parse error.

**Comment by mhegazy** [comment]
@sandersn, we also need to ensure that applicable quick fixes are working correctly in JSDoc, e.g. https://github.com/Microsoft/TypeScript/issues/15852

**Comment by andy-ms** [comment]
Note: `fixSpelling.ts` would need to be updated so that `getTokenAtPosition` uses `/*includeJsDocComment*/ true`. (But that alone wouldn't be sufficent.)

**Comment by sandersn** [comment]
@mhegazy @andy-ms that quick fix is for a semantic error. That's a separate issue from this one which reports syntax errors. I'll re-open #15852 to track that.

You're not talking about the quick fix "Disable checking for this file", right?

**Comment by sandersn** [comment]
Fix is up at #16156

**Comment by andy-ms** [comment]
Just got tripped up by this:
```js
// @ts-check

/** @typedef Foo {number} */

/** @type Foo */
const x = "";
```

This isn't an error today. Could we report an error on a `@typedef` tag where we didn't parse any type?

**Comment by Kovensky** [comment]
The error there’s is actually on the `@type` tag rather than the `@typedef`. But yes, sounds reasonable to raise an error.

**Comment by andy-ms** [comment]
@Kovensky Thanks, that is actually 2 errors in one. #16946 #16947
